### PR TITLE
Prepare 1.44.2-dev.0001

### PR DESCRIPTION
**Pull request by JaniruTEC**
This PR prepares the internal release 1.44.2-dev.0001.

## Summary by CodeRabbit

- **New Features**
	- Introduced GitHub action workflows for Java 7 core library testing and Renovate bot config validation.
	- Added support for Java 21 in CI build and test process.
	- Implemented `isShutdown()` method to check if transport is shutdown in `HttpTransport`.
- **Bug Fixes**
	- Updated URL in documentation link to HTTPS.
	- Fixed `setValue` method and `settersMethodForField` logic in `FieldInfo`.
	- Corrected URI Templates reserved expansion in `PercentEscaper`.
- **Documentation**
	- Updated README to reflect the creation of an unofficial fork for Cryptomator for Android.
	- Updated CHANGELOG for version 1.44.1 focusing on bug fixes, dependency updates, and new features.
- **Refactor**
	- Modified file header in `HttpRequest` to add notice of modification by Cryptomator team.
- **Chores**
	- Updated dependency configurations, Docker image versions, and added security vulnerability notifications setup.
	- Configured URL protocols for `http` and `https` in native-image properties.

### CODE REVIEW

**Comment by JaniruTEC** [comment]
@coderabbitai pause

**Comment by JaniruTEC** [comment]
@ People from Google:
I apologize for the inconvenience if you got a notification for this PR. Our CLA Bot was misconfigured and pinged all of you erroneously.
No action is required and it won't happen again. Sorry!

File: README.md [hunk 1]
```diff
@@ -1,4 +1,17 @@
...
+> For a project designed for use by the public, please refer to the [original repository.](https://github.com/googleapis/google-http-java-client)
+>
+> The README of the original project is reproduced below for your convenience and to provide proper attribution.  
+> This project is not affiliated, associated, endorsed by, or in any way officially connected with Google, or any of its subsidiaries or its affiliates.
+
+---
+
+# Original README
 
 [![Maven][maven-version-image]][maven-version-link]
```
Comment (SailReal) [inline README.md]: Would also remove those three lines

**Review by SailReal — COMMENTED** [review]
LGTM :+1: , would only remove those images in the README as well

File: README.md (hunk 1 above)
Comment (JaniruTEC) [reply README.md]: [Line 12](https://github.com/cryptomator/google-http-java-client/pull/11/files/adf0aa5d6a558704abb99c890592d95ddf322499..f7865b43e859d527b8b13d44f6c12ddae0fb2f95#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R12) is used to insert a horizontal divider in the rendered README.
The badges declared in [lines 16-18](https://github.com/cryptomator/google-http-java-client/pull/11/files/adf0aa5d6a558704abb99c890592d95ddf322499..f7865b43e859d527b8b13d44f6c12ddae0fb2f95#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R16-R18) have been removed by 95aa6d98df9cd98c7f5ae4a21c7f08e7c8ff65c6.

**Pull request merged**